.github/workflows/e2e-nvidia-l40s-x4.yml

```
@@ -170,8 +170,25 @@ jobs:
           HF_TOKEN: ${{ secrets.HF_TOKEN }}
         run: |
           . venv/bin/activate
+
+          # set preserve to true so we can retain the logs
+          export PRESERVE=1
           ./scripts/e2e-ci.sh -l
 
+          # we know that the file will be named something like f"/training_params_and_metrics_global{os.environ['RANK']}.jsonl" in python
+          # and we know that it will be written into a directory created by mktemp -d. 
```

Same comment as above

```suggestion
          # and we know that it will be written into a directory created by `mktemp -d`
```

---

I made the change, thank you for paying attention to details like this. It makes our codebase more enjoyable to work on in the long-term.